UniAtHome/UniAtHome.WebAPI/Controllers/UsersController.cs

```diff
@@ -0,0 +1,100 @@
+﻿using Microsoft.AspNetCore.Authentication.JwtBearer;
+using Microsoft.AspNetCore.Authorization;
+using Microsoft.AspNetCore.Identity;
+using Microsoft.AspNetCore.Mvc;
+using System.Collections;
+using System.Linq;
+using System.Security.Claims;
+using System.Threading.Tasks;
+using UniAtHome.BLL.Interfaces;
+using UniAtHome.DAL.Entities;
+using UniAtHome.WebAPI.Models.UsersRequests;
+
+namespace UniAtHome.WebAPI.Controllers
+{
+    [Route("api/[controller]")]
+    [ApiController]
+    public class UsersController : ControllerBase
+    {
+        private readonly UserManager<User> userManager;
+
+        private readonly IAuthTokenGenerator tokenGenerator;
+
+        public UsersController(UserManager<User> userManager, IAuthTokenGenerator tokenGenerator)
+        {
+            this.userManager = userManager;
+            this.tokenGenerator = tokenGenerator;
+        }
+
+        [AllowAnonymous]
+        [HttpPost("register")]
+        public async Task<object> Register([FromBody] RegistrationRequest registerModel)
+        {
+            var user = new User
+            {
+                UserName = registerModel.Email,
+                FirstName = registerModel.FirstName,
+                LastName = registerModel.LastName,
+                Email = registerModel.Email
+            };
+            var registerResult = await userManager.CreateAsync(user, registerModel.Password);
+            if (registerResult.Succeeded)
+            {
+                return Ok("User is successfully registered!");
+            }
+
+            return GetErrorOfNotSucceededRequest(registerResult);
+        }
+
+        private ObjectResult GetErrorOfNotSucceededRequest(IdentityResult registerResult)
```

Comment:
Why do you use ObjectResult instead ActionResult? What diference between them?

Reply:
I thought ObjectResult takes into account user's response preferences (XML, Json) and serializes returned object this way.